core/templates/cowdata.h

```diff
@@ -64,45 +64,74 @@ class CowData {
 	template <class TV, class VV>
 	friend class VMap;
 
+public:
+#ifdef IS_32_BIT
+	typedef int32_t Size;
+	typedef uint32_t USize;
+#else
+	typedef int64_t Size;
+	typedef uint64_t USize;
+#endif
 private:
+	// Function to find the next power of 2 to an integer.
+	static _FORCE_INLINE_ USize next_po2(USize x) {
+		if (x == 0) {
+			return 0;
+		}
+
+		--x;
+		x |= x >> 1;
+		x |= x >> 2;
+		x |= x >> 4;
+		x |= x >> 8;
+		x |= x >> 16;
+		if (sizeof(USize) == 8) {
+			x |= x >> 32;
```

Comment:
This leaves the maximum size as `2^33`, is this intended as a limit to not go too big or an oversight?